src/telegram.ts

```diff
@@ -89,23 +89,35 @@ class Telegram extends ApiClient {
    * @param certificate Upload your public key certificate so that the root certificate in use can be checked
    * @param maxConnections Maximum allowed number of simultaneous HTTPS connections to the webhook for update delivery, 1-100
    * @param allowedUpdates List the types of updates you want your bot to receive
+   * @param ipAddress The fixed IP address which will be used to send webhook requests instead of the IP address resolved through DNS
+   * @param dropPendingUpdates Whether to drop pending updates
    */
   setWebhook(
     url: string,
     certificate?: tt.InputFile,
     maxConnections?: number,
-    allowedUpdates?: readonly tt.UpdateType[]
+    allowedUpdates?: readonly tt.UpdateType[],
+    ipAddress?: string,
+    dropPendingUpdates?: boolean
```

Comment:
Optional arguments (especially `boolean`) should be passed as `extra` object.

Reply:
Should I move all optional aruments to `extra`, or only the new ones?

Reply:
Your call, both approaches have merit

Reply:
Move all.